packages/aws-cdk/lib/api/hotswap/common.ts

```diff
@@ -33,6 +34,43 @@ export enum ChangeHotswapImpact {
 
 export type ChangeHotswapResult = HotswapOperation | ChangeHotswapImpact;
 
-export function assetMetadataChanged(change: cfn_diff.ResourceDifference): boolean {
+/**
+ * Represents a change that can be hotswapped.
+ */
+export class HotswappableResourceChange {
```

Comment:
I feel like this name is not great, because we actually don't know whether this change is hotswappable yet (the detector functions will determine that).

How about `HotswappableChangeCandidate`?